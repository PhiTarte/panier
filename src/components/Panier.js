// TODO Panier qui affiche le panier
import React, { Component } from 'react'
import { Button, Container } from 'reactstrap'
import ListArticle from './ListArticle'

class Panier extends Component {

    constructor() {
        super()
        //state initial
        this.state = {
            articles: [
                {
                    name:"Zidane",
                    price:"45",
                    imgSrc:"https://www.foot01.com/img/images/650x600/2018/Jan/20/benzema-revient-zidane-recupere-l-arme-fatale-du-real-zidane-16,207169.jpg"
                },
                {
                    name:"Leboeuf",
                    price:"45",
                    imgSrc:"https://www.foot01.com/img/images/650x600/2018/Jan/20/benzema-revient-zidane-recupere-l-arme-fatale-du-real-zidane-16,207169.jpg"
                },
                {
                    name:"Pires",
                    price:"45",
                    imgSrc:"https://www.foot01.com/img/images/650x600/2018/Jan/20/benzema-revient-zidane-recupere-l-arme-fatale-du-real-zidane-16,207169.jpg"
                },
                {
                    name:"Karembeu",
                    price:"45",
                    imgSrc:"https://www.foot01.com/img/images/650x600/2018/Jan/20/benzema-revient-zidane-recupere-l-arme-fatale-du-real-zidane-16,207169.jpg"
                }
              ]
        }
    }

    render() {
        return <Container>
            <ListArticle articles={this.state.articles}/>
            <Button />
        </Container>
    }
}

export default Panier
