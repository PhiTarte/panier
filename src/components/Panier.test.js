import React from 'react';
import ReactDOM from 'react-dom';
import Panier from './Panier';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Panier />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders Panier html', () => {
 
  const tree = renderer
    .create(<Panier />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
