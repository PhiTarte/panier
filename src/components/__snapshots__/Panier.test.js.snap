// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Panier html 1`] = `
<div
  className="container"
>
  <ul
    className="list-group"
  >
    <li
      className="list-group-item"
    >
      <div
        className="row"
      >
        <div
          className="col-3"
        >
          <button
            className="btn btn-secondary"
            onClick={[Function]}
            type={undefined}
          >
            Supp
          </button>
        </div>
        <div
          className="col-3"
        >
          <img
            alt="Zidane"
            className="img-fluid"
            src="https://www.foot01.com/img/images/650x600/2018/Jan/20/benzema-revient-zidane-recupere-l-arme-fatale-du-real-zidane-16,207169.jpg"
          />
        </div>
        <div
          className="col-3"
        >
          <h3>
            Zidane
          </h3>
        </div>
        <div
          className="col-3"
        >
          <p>
            45
             €
          </p>
        </div>
      </div>
    </li>
    <li
      className="list-group-item"
    >
      <div
        className="row"
      >
        <div
          className="col-3"
        >
          <button
            className="btn btn-secondary"
            onClick={[Function]}
            type={undefined}
          >
            Supp
          </button>
        </div>
        <div
          className="col-3"
        >
          <img
            alt="Leboeuf"
            className="img-fluid"
            src="https://www.foot01.com/img/images/650x600/2018/Jan/20/benzema-revient-zidane-recupere-l-arme-fatale-du-real-zidane-16,207169.jpg"
          />
        </div>
        <div
          className="col-3"
        >
          <h3>
            Leboeuf
          </h3>
        </div>
        <div
          className="col-3"
        >
          <p>
            45
             €
          </p>
        </div>
      </div>
    </li>
    <li
      className="list-group-item"
    >
      <div
        className="row"
      >
        <div
          className="col-3"
        >
          <button
            className="btn btn-secondary"
            onClick={[Function]}
            type={undefined}
          >
            Supp
          </button>
        </div>
        <div
          className="col-3"
        >
          <img
            alt="Pires"
            className="img-fluid"
            src="https://www.foot01.com/img/images/650x600/2018/Jan/20/benzema-revient-zidane-recupere-l-arme-fatale-du-real-zidane-16,207169.jpg"
          />
        </div>
        <div
          className="col-3"
        >
          <h3>
            Pires
          </h3>
        </div>
        <div
          className="col-3"
        >
          <p>
            45
             €
          </p>
        </div>
      </div>
    </li>
    <li
      className="list-group-item"
    >
      <div
        className="row"
      >
        <div
          className="col-3"
        >
          <button
            className="btn btn-secondary"
            onClick={[Function]}
            type={undefined}
          >
            Supp
          </button>
        </div>
        <div
          className="col-3"
        >
          <img
            alt="Karembeu"
            className="img-fluid"
            src="https://www.foot01.com/img/images/650x600/2018/Jan/20/benzema-revient-zidane-recupere-l-arme-fatale-du-real-zidane-16,207169.jpg"
          />
        </div>
        <div
          className="col-3"
        >
          <h3>
            Karembeu
          </h3>
        </div>
        <div
          className="col-3"
        >
          <p>
            45
             €
          </p>
        </div>
      </div>
    </li>
  </ul>
  <button
    className="btn btn-secondary"
    onClick={[Function]}
    type={undefined}
  />
</div>
`;
